src/Shared/Cosmos/CosmosConstants.cs

```diff
@@ -5,6 +5,12 @@ namespace Aspire.Hosting.Azure.Cosmos;
 
 internal static class CosmosConstants
 {
+    /// <summary>
+    /// Defines the application name used to interact with the aszure cosmos db. This will be suffixed to the
+    /// cosmos user-agent to include with every Azure Cosmos DB service interaction.
+    /// </summary>
+    internal const string CosmosApplicationName = "AspireCosmosDBClient";
```

Comment:
Is there an upper limit on the User Agent? If so, we might want to consider shortening this if possible. Maybe just `Aspire` as the User Agent already contains the fact that it's a Cosmos DB Client.

Reply:
We already have a check in place for validating and trimming the `Suffix` within the `Microsoft.Azure.Documents.UserAgentContainer`:

```
    public string Suffix
    {
        get
        {
            return suffix;
        }
        set
        {
            suffix = value;
            if (suffix.Length > 64)
            {
                suffix = suffix.Substring(0, 64);
            }

            userAgent = BaseUserAgent + suffix;
            userAgentUTF8 = Encoding.UTF8.GetBytes(userAgent);
        }
    }
```

So, in case it's larger than the defined size, we will fit it within the `64` character window, which would still contain the required application name prefix for us to track.

Reply:
That's exactly the point. Because there is a limit, any character we used for the fixed part takes away from the potential user app name.

Reply:
I would agree with @ealsur. Just by using `Aspire` doesn't make much difference, in terms of identification. With this, we would make extra room for the user provided app name, if any.

Addressed with a new commit.